content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1056 @@
+  }
+)
+
+const UserSchema = new Schema({
+  name: String,
+  email: {
+    type: String,
+    unique: true,
+  },
+  profile: {
+    type: ProfileSchema,
+    default: () => ({}),
+  },
+})
+
+module.exports = mongoose.model('User', UserSchema)
+```
+
+</tab>
+
+<tab>
+
+```js
+const mongoose = require('mongoose')
+
+const Schema = mongoose.Schema
+
+const CategorySchema = new Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+})
+
+module.exports = mongoose.model('Category', CategorySchema)
+```
+
+</tab>
+
+</TabbedContent>
+
+The models/documents have the following relationships:
+
+- 1-n: `User` ↔ `Post`
+- m-n: `Post` ↔ `Category`
+- Sub-document/ Embedded document: `User` ↔ `Profile`
+
+In this guide, the route handlers are located in the `src/controllers` directory. The models are located in the `src/models` directory. From there, they are pulled into a central `src/routes.js` file which is used to set up the required routes in `src/index.js`:
+
+```copy=false
+└── blog-mongoose
+    ├── package.json
+    └──src
+        ├── controllers
+        │   ├── post.js
+        │   └── user.js
+        ├── models
+        │   ├── category.js
+        │   ├── post.js
+        │   └── user.js
+        ├── index.js
+        ├── routes.js
+        └── seed.js
+```
+
+The example repository contains a `seed` script inside the `package.json` file. Run `npm run seed` to populate your database with the sample data in the `./src/seed.js` file.
+
+## Step 1. Install the Prisma CLI
+
+The first step to adopt Prisma is to [install the Prisma CLI](/concepts/components/prisma-cli/installation#local-installation-recommended) in your project:
+
+```terminal copy
+npm install prisma --save-dev
+```
+
+## Step 2. Introspect your database
+
+### 2.1. Set up Prisma
+
+Before you can introspect your database, you need to set up your [Prisma schema](/concepts/components/prisma-schema) and connect Prisma to your database. Run the following command in your terminal to create a basic Prisma schema file:
+
+```terminal copy
+npx prisma init
+```
+
+This command created:
+
+- A new directory called `prisma` with a `schema.prisma`: Your Prisma schema file that specifies your database connection and models
+- `.env`: A [`dotenv`](https://github.com/motdotla/dotenv) at the root of your project, if it doesn't exist, to configure your database connection URL as an environment variable
+
+The Prisma schema file currently looks as follows:
+
+```prisma file=prisma/schema.prisma
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+datasource db {
+  provider = "mongodb"
+  url      = env("DATABASE_URL")
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+```
+
+<Tip>
+
+If you're using VS Code, be sure to install the [Prisma VS Code extension](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma) for syntax highlighting, formatting, auto-completion and a lot more cool features.
```

Comment:
Maybe we go for a more general call and point users to check this Editor extensions page https://www.prisma.io/docs/guides/development-environment/editor-setup ?

Reply:
Additionally maybe, VS Code will get 90% of users.